File: apps/mobile/lib/models/candidate_profile.dart
class CandidateProfile {
  final String id;
  final String fullName;
  final String? phone;
  final String city;
  final String? district;
  final int experienceYears;
  final List<String> skills;
  final List<String> workPreferences;
  final String? primaryTradeCategory;
  final bool isPublic;
  final String availabilityStatus;

  const CandidateProfile({
    required this.id,
    required this.fullName,
    this.phone,
    required this.city,
    this.district,
    required this.experienceYears,
    required this.skills,
    required this.workPreferences,
    this.primaryTradeCategory,
    required this.isPublic,
    this.availabilityStatus = 'AVAILABLE',
  });

  factory CandidateProfile.fromJson(Map<String, dynamic> json) => CandidateProfile(
        id: json['id'] as String,
        fullName: json['fullName'] as String,
        phone: json['phone'] as String?,
        city: json['city'] as String,
        district: json['district'] as String?,
        experienceYears: json['experienceYears'] as int,
        skills: (json['skills'] as List?)?.map((e) => e as String).toList() ?? [],
        workPreferences: (json['workPreferences'] as List?)?.map((e) => e as String).toList() ?? [],
        primaryTradeCategory: json['primaryTradeCategory'] as String?,
        isPublic: json['isPublic'] as bool? ?? false,
        availabilityStatus: json['availabilityStatus'] as String? ?? 'AVAILABLE',
      );
}

File: packages/shared/src/types/candidate-profile.ts
import { AvailabilityStatus } from "./enums";

export interface CandidateProfileDto {
  id: string;
  fullName: string;
  phone?: string | null;
  phoneVisible: boolean;
  city: string;
  district?: string | null;
  address?: string | null;
  bio?: string | null;
  experienceYears: number;
  skills: string[];
  workPreferences: string[];
  primaryTradeCategory?: string | null;
  isPublic: boolean;
  photoUrl?: string | null;
  photoVisible: boolean;
  availabilityStatus: AvailabilityStatus;
  /** Sadece BUSY durumundayken anlamlı — hangi tarihten itibaren müsait olunacağı. */
  availableFrom?: string | null;
  notifyByEmail: boolean;
  hasProfessionalQualificationCert: boolean;
  hasMasterCraftsmanCert: boolean;
  hasOshCert: boolean;
  serviceRadiusKm?: number | null;
}

export interface UpdateCandidateProfileInput {
  fullName?: string;
  phone?: string;
  phoneVisible?: boolean;
  city?: string;
  district?: string;
  address?: string;
  bio?: string;
  experienceYears?: number;
  skills?: string[];
  workPreferences?: string[];
  primaryTradeCategory?: string;
  isPublic?: boolean;
  photoVisible?: boolean;
  availabilityStatus?: AvailabilityStatus;
  availableFrom?: string;
  notifyByEmail?: boolean;
  hasProfessionalQualificationCert?: boolean;
  hasMasterCraftsmanCert?: boolean;
  hasOshCert?: boolean;
  serviceRadiusKm?: number;
}
